.github/workflows/e2e-tests-embedding-sdk.yml

```
@@ -0,0 +1,204 @@
+name: E2E Cross-version Tests for Embedding SDK
+
+on:
+  push:
+    branches:
+      - "master"
+      - "release-**"
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  files-changed:
+    name: Check which files changed
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
+    outputs:
+      e2e_all: ${{ steps.changes.outputs.e2e_all }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test which files changed
+        uses: dorny/paths-filter@v3.0.0
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-paths.yaml
+
+  # if this is a test on a release branch, we need to check the build requirements
+  get-build-requirements:
+    if: |
+      !cancelled()
```

nit: this is short, so I'd make them on the same line
```suggestion
    if: !cancelled()
```

But the question is why do we need `!cancelled()`? [From the doc](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#always), it seems this is for when we want to always run the step regardless of the step failure. I don't know why we want to do that here.

I traced this back to https://github.com/metabase/metabase/pull/28171 which introduced `always()` which later is changed to `!cancelled()` in https://github.com/metabase/metabase/pull/32951.

@alxnddr why do we need `!cancelled()` (previously `always()`) in the E2E workflow?

---

oh, the [doc actually suggests](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#always) `if: ${{ !cancelled() }}`. I completely missed that. I'd suggest, if we have a chance to touch this file again, to follow GitHub's doc for consistency.